icu4j/main/classes/core/src/com/ibm/icu/impl/CaseMapImpl.java

```diff
@@ -794,6 +794,10 @@ public static String toTitle(int caseLocale, int options, BreakIterator iter, Ch
     private static <A extends Appendable> int maybeTitleDutchIJ(
             CharSequence src, int c, int start, int segmentLimit,
             A dest, int options, Edits edits) throws IOException {
+        if (start >= segmentLimit) {
```

Comment:
You should not need this, assuming it works like the UTF-16 code in C++.

optional:
If you like, you could
```
assert start < segmentLimit;
```
(and U_ASSERT(...) in C++)

Reply:
Done.